swap/swap.go

```diff
@@ -59,15 +63,38 @@ func (s *Swap) Add(amount int64, peer *protocols.Peer) (err error) {
 	if err != nil && err != state.ErrNotFound {
 		return
 	}
+
+	//check if balance with peer is over the disconnect threshold
+	if s.balances[peer.ID()] >= s.disconnectThreshold {
+		//if so, return error in order to abort the transfer
+		return fmt.Errorf("balance for peer %s went over the disconnect threshold %v", peer.ID().String(), s.disconnectThreshold)
+	}
+
 	//adjust the balance
 	//if amount is negative, it will decrease, otherwise increase
 	s.balances[peer.ID()] += amount
+
 	//save the new balance to the state store
 	peerBalance := s.balances[peer.ID()]
 	err = s.stateStore.Put(peer.ID().String(), &peerBalance)
+	if err != nil {
+		return
+	}
 
 	log.Debug(fmt.Sprintf("balance for peer %s: %s", peer.ID().String(), strconv.FormatInt(peerBalance, 10)))
-	return err
+
+	//check if balance with peer is over the payment threshold
+	if peerBalance >= s.paymentThreshold {
+		//if so, send cheque request message with the current peer balance as its amount
+		err = s.ChequeRequestMsg(peer, peerBalance)
+	}
+
+	return
+}
+
+func (s *Swap) ChequeRequestMsg(peer *protocols.Peer, amount int64) error {
```

Comment:
This function is not needed as the message is sent as `peer.Send(ctx, &ChequeRequestMsg{})`

Reply:
Very well, I will replace this with the proper call.